src/redisearch_rs/low_memory_thin_vec/src/lib.rs

```diff
@@ -1313,6 +1352,52 @@ impl<T: Clone> LowMemoryThinVec<T> {
     }
 }
 
+impl<T: Copy + std::fmt::Debug> LowMemoryThinVec<T> {
+    /// Reserves capacity to fit the given `prefix` slice,
+    /// moves the current elements back to make room for the prefix
+    /// and copies the prefix to the front of the vector.
+    pub fn prepend_with_slice(&mut self, prefix: &[T]) {
+        let prefix_len = prefix.len();
+        let self_len = self.len();
+        let new_len = prefix_len + self_len;
+        debug_assert!(new_len <= isize::MAX as usize);
+
+        if prefix.is_empty() {
+            return;
+        }
+
+        if self.is_empty() {
+            self.extend_from_slice(prefix);
+            return;
+        }
+
+        self.reserve(prefix_len);
+        {
+            // SAFETY:
+            // We reserved enough space for an additional `prefix_len`
+            // amount of elements.
+            let dst = unsafe { self.data_raw().add(prefix_len) };
+            // SAFETY:
+            // We have reserved enough space for `prefix_len` elements,
+            // so it is safe to copy the elements from the current vector to the
+            // newly reserved space.
+            unsafe { std::ptr::copy(self.data_raw(), dst, self_len) };
+        }
+
+        // SAFETY:
+        // `prefix` cannot be a reference to a slice in `self`,
+        // as that would violate borrowing rules. Therefore, we can safely
+        // memcpy the elements from `prefix` to start of the buffer.
```

Comment:
We should note that this is only safe due to the fact that `T` is `Copy`.

Reply:
Fixed in https://github.com/RediSearch/RediSearch/commit/4ca45094e575cd88387bc59e3fa7918201237a8e.